src/nms.py

```diff
@@ -137,26 +142,27 @@ def _make_request(
             )
         except requests.exceptions.SSLError as e:
             logger.error("SSL error: %s", e)
-            return None
+            raise NMSError() from e
         except requests.RequestException as e:
             logger.error("HTTP request failed: %s", e)
-            return None
+            raise NMSError() from e
         except OSError as e:
             logger.error("couldn't complete HTTP request: %s", e)
-            return None
+            raise NMSError() from e
         try:
             response.raise_for_status()
-        except requests.HTTPError:
+        except requests.HTTPError as e:
             logger.error(
-                "%s request failed: code %s",
+                "%s request failed: code %s with message: %s",
                 endpoint,
                 response.status_code,
+                response.text,
             )
-            return None
+            raise NMSError() from e
         try:
             json_response = response.json()
-        except json.JSONDecodeError:
-            return None
+        except json.JSONDecodeError as e:
+            raise NMSError() from e
         return json_response
 
     def is_initialized(self) -> bool:
```

Comment:
This should `return status.get("initialized")`

Reply:
Same as above